python/pyspark/daemon.py

```diff
@@ -93,8 +94,19 @@ def manager():
     # Create a new process group to corral our children
     os.setpgid(0, 0)
 
+    is_unix_domain_sock = os.environ.get("PYTHON_UNIX_DOMAIN_ENABLED", "false").lower() == "true"
+    socket_path = None
+
     # Create a listening socket on the loopback interface
-    if os.environ.get("SPARK_PREFER_IPV6", "false").lower() == "true":
+    if is_unix_domain_sock:
+        socket_path = os.path.join(pathlib.Path.home(), f".{uuid.uuid4()}.sock")
+        if os.path.exists(socket_path):
+            os.remove(socket_path)
+        listen_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
```

Comment:
UDS is now supported in Windows too FWIW (https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/)